### PR TITLE
add properties to project and allocation classes

### PR DESCRIPTION
**Pull request by zhenz**

### CODE REVIEW

File: pytas/models/projects.py [hunk 1]
```diff
@@ -140,6 +179,9 @@ def percentComputeUsed(self):
             return (used / alloc) * 100
         return 0
 
+    @property
+    def days_left(self):
+        return (self.end - datetime.today()).days
```
Comment (diurnalist) [inline pytas/models/projects.py]: What about timezone?

**Review by diurnalist — APPROVED** [review]
These are great abstractions that will help make the calling code much cleaner! Thanks.

File: pytas/models/projects.py [hunk 2]
```diff
@@ -93,7 +94,45 @@ def remove_user(self, username):
         api = TASClient()
         return api.del_project_user(self.id, username)
 
+    @property
+    def active_allocations(self):
```
Comment (xandroc) [inline pytas/models/projects.py:100]: I get an error when using these changes to view a project in portal while using master. It works if I use the portal's allocation branch. I think the issue is we try and set this property in the portal codebase but the property on the pytas side doesn't have a setter https://github.com/ChameleonCloud/portal/blob/master/projects/views.py#L162

File: pytas/models/projects.py [hunk 3]
```diff
@@ -140,7 +181,18 @@ def percentComputeUsed(self):
...
+        return (self.end - datetime.utcnow()).days
 
+    @property
+    def up_for_renewal(self):
+        days_left = self.days_left
+        return days_left >= 0 and days_left <= RENEWAL_START_WINDOW
+
+    @property
+    def renewal_days(self):
+        return self.days_left
```
Comment (diurnalist) [inline pytas/models/projects.py:195]: Arguably this should be `None` if `up_for_renewal` is `False`. Or, we could just not have it. But, this is fine :)

**Review by diurnalist — APPROVED** [review]
Great changes! :shipit:

File: pytas/models/projects.py (hunk 2 above)
Comment (diurnalist) [reply pytas/models/projects.py:100]: @zhenz this means we'll have to merge both this and the Portal PR at the same time, unless you change the requirements.txt of Portal to point to a pytas branch with these changes. They need to be rolled out at the same time.

**Pull request merged**